rclpy/src/rclpy/_rclpy.c

```diff
@@ -1441,7 +1441,7 @@ rclpy_create_service(PyObject * Py_UNUSED(self), PyObject * args)
       PyErr_Format(PyExc_RuntimeError,
         "Failed to create service: %s", rcl_get_error_string_safe());
     }
-    Py_DECREF(service);
+    Py_DECREF(pyservice);
```

Comment:
Is this also necessary before returning in line 1430?

Does `service` need to be deallocated?

Reply:
I think this should be a follow up PR. We should check that every creation that has been moved up a few blocks is properly deallocated in case of failure

Reply:
Yes to both. For the first one the `PyCapsule_New` line could be moved after 1430.

Reply:
So do we agree this can be merged as is ?
@sloretz would you mind submitting a separate PR to address the remaining items ?

Reply:
@mikaelarguedas WIll do

Reply:
#138